src/app/session/active/[templateId]/page.tsx

```
@@ -85,28 +124,62 @@ export default function ActiveSessionPage({
+  }, [updateWorkoutMutation]);
+
+  const toggleTimer = useCallback(() => {
+    if (activeWorkout) {
+      updateWorkoutMutation.mutate({
+        isTimerActive: !activeWorkout.isTimerActive,
+        lastPauseTime: !activeWorkout.isTimerActive ? undefined : new Date().toISOString(),
+      });
+    }
+  }, [activeWorkout, updateWorkoutMutation]);
+
+  const updateExerciseProgress = useCallback((exerciseId: string, progress: any) => {
```

ESLint

[error] React Hook "useCallback" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?

(react-hooks/rules-of-hooks)